code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -36,10 +36,16 @@ emp_act
 		P.on_hit(src, 100, def_zone)
 		return 2
 
-	if(mind?.martial_art?.deflection_chance) //Some martial arts users can deflect projectiles!
+	if(mind?.martial_art?.deflection_chance && in_throw_mode) //Some martial arts users can deflect projectiles!
```

Comment:
What if we want some martial arts to reflect things while not in throw mode? Perhaps have a proc on the martial art called `try_deflect()` or whatever, that returns the `prob(deflection chance)` by default. And then carp would additionally return whether they were in throw mode.

Reply:
now you all martial arts deflect normally however sleeping carp only deflects on throw mode